recipes/libqasm/all/conanfile.py

```diff
@@ -0,0 +1,124 @@
+        self.options["antlr4-cppruntime"].shared = self.options.shared
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def build_requirements(self):
+        self.tool_requires("tree-gen/1.0.7")
+        self.tool_requires("zulu-openjdk/11.0.19")
+        if self.settings.arch == "wasm":
+            self.tool_requires("emsdk/3.1.50")
+        if self._should_build_test:
+            self.test_requires("gtest/1.14.0")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._min_cppstd},"
+                                            f"which your compiler does not support.")
+
+        if self.dependencies["antlr4-cppruntime"].options.shared != self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} requires antlr4-cppruntime to be built with the same shared option value.")
+
+    def requirements(self):
+        self.requires("fmt/10.2.1", transitive_headers=True)
+        self.requires("range-v3/0.12.0", transitive_headers=True)
+        self.requires("tree-gen/1.0.7", transitive_headers=True, transitive_libs=True)
+        if not self.settings.arch == "wasm":
+            self.requires("antlr4-cppruntime/4.13.1", transitive_headers=True)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        deps = CMakeDeps(self)
+        deps.generate()
+        tc = CMakeToolchain(self)
+        tc.variables["LIBQASM_BUILD_PYTHON"] = self.options.build_python
+        tc.variables["LIBQASM_BUILD_TESTS"] = self._should_build_test
+        tc.generate()
+        env = VirtualBuildEnv(self)
+        env.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE.md", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        fix_apple_shared_install_name(self)
+        rm(self, "*.interp", os.path.join(self.package_folder, "include"), recursive=True)
+        rm(self, "*.pdb", os.path.join(self.package_folder), recursive=True)
+        rm(self, "*.tokens", os.path.join(self.package_folder, "include"), recursive=True)
+        rm(self, "cmake_install.cmake", os.path.join(self.package_folder, "include"), recursive=True)
+        rm(self, "Makefile", os.path.join(self.package_folder, "include"), recursive=True)
+        rmdir(self, os.path.join(os.path.join(self.package_folder, "include", "CMakeFiles")))
+        rmdir(self, os.path.join(os.path.join(self.package_folder, "include", "v3x", "CMakeFiles")))
```

Comment:
Are the CMake install() commands in the project buggy? These look out of place, especially in the `include` dir.

Reply:
Yes, that's something I definitely have to have a look at, because I've seen my IDE, locally, with these CMake files and folders all around. Thanks!

Reply:
Looked into it. The CMake install command needs some tweaking to exclude irrelevant files from the build dir.
```patch
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -315,9 +315,11 @@
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 )
-get_target_property(INCLUDE_DIRS cqasm-lib-obj INTERFACE_INCLUDE_DIRECTORIES)
 install(
-    DIRECTORY ${INCLUDE_DIRS}
+    DIRECTORY
+        "${CMAKE_CURRENT_SOURCE_DIR}/../include/"
+        "${CMAKE_CURRENT_BINARY_DIR}/../include/v3x"
+        "${CMAKE_CURRENT_BINARY_DIR}/v3x"
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
-    PATTERN "*.hpp" PATTERN "*.inc"
+    FILES_MATCHING PATTERN "*.hpp" PATTERN "*.h" PATTERN "*.inc"
 )
```
I would also recommend moving the generated headers from `${CMAKE_CURRENT_BINARY_DIR}/v3x` to `${CMAKE_CURRENT_BINARY_DIR}/include/v3x`, if possible, since it's still creating empty `CMakeFiles` in the installed `include/` dir.

Reply:
Many thanks for the review, @valgur! I think I will  put all of these changes into a libqasm/0.6.4.

Reply:
I have the changes ready to be tested next week in libqasm. I have updated the script that is invoking ANTLR to generate the lexer and parser files so that it moves the `.h` files to `include/v3x`. With that change, we wouldn't need that `${CMAKE_CURRENT_BINARY_DIR}/v3x` line anymore.